src/main/java/uk/org/tombolo/importer/phe/ChildhoodObesityImporter.java

```diff
@@ -98,113 +72,181 @@ public DatasourceSpec getDatasourceSpec(String datasourceId) throws Exception {
-                    subjectType = SubjectTypeUtils.getOrCreate(AbstractONSImporter.PROVIDER,
-                            OaImporter.OaType.localAuthority.name(), OaImporter.OaType.localAuthority.datasourceSpec.getDescription());
-                    break;
-                case msoa:
-                    sheet = workbook.getSheet("MSOAData_2011-12_2013-14");
-                    subjectType = SubjectTypeUtils.getOrCreate(AbstractONSImporter.PROVIDER,
-                            OaImporter.OaType.msoa.name(), OaImporter.OaType.msoa.datasourceSpec.getDescription());
-                    break;
-                case ward:
-                    throw new Error("Wards are not yet supported");
-                    // FIXME: In case we want to suport wards at some point, here is the sheet to use
-                    //sheet = workbook.getSheet("WardData_2011-12_2013-14");
-                    //break;
+
+        // Check the datasourceid to reference the corresponding url
+        String which_datasource = datasource.getDatasourceSpec().getId();
+        String fileLocation = getDatasourceSpec(which_datasource).getUrl();
+
+        List<TimedValue> timedValues = new ArrayList<TimedValue>();
+
+        // The code below fetches the .xls file from the URL we specified in our DatasourceSpec object
+        URL url;
+        try {
+            url = new URL(fileLocation);
+        } catch (MalformedURLException e) {
+            File file;
+            if (!(file = new File(fileLocation)).exists()) {
+                log.error("File does not exist: " + fileLocation);
             }
-            if (sheet == null)
-                throw new Error("Sheet not found for datasource: " + datasource.getDatasourceSpec().getId());
+            url = file.toURI().toURL();
+        }
+
+        // Fetching and reading the file using fetchInputStream
+        InputStream isr = downloadUtils.fetchInputStream(url, getProvider().getLabel(), ".xlsx");
+        XSSFWorkbook workbook = new XSSFWorkbook(isr);
+        DataFormatter dataFormatter = new DataFormatter();
+
+
+        // Loop through the excell sheets
+        for (int sheet = 1; sheet <= 4; sheet = sheet+1){
+            Sheet datatypeSheet = workbook.getSheetAt(sheet);
 
-            // Define a list of timed value extractor, one for each attribute
-            List<TimedValueExtractor> timedValueExtractors = new ArrayList<>();
+            Iterator<Row> rowIterator = datatypeSheet.rowIterator();
+
+            // Dataset specific: this is to skip the first two lines that don't have any values of interest
+            if (which_datasource.equals("childhoodObesityLA") || which_datasource.equals("childhoodObesityMSOA")){
+                rowIterator.next();
+                rowIterator.next();
+                rowIterator.next();
+            } else {
+                rowIterator.next();
+                rowIterator.next();
+                rowIterator.next();
+                rowIterator.next();
+            }
```

Comment:
could be written as

```
int j = which_datasource.equals("childhoodObesityLA") 
                        || which_datasource.equals("childhoodObesityMSOA") ? 3 : 4;
for (int i=0; i<j; i++){
    if(rowIterator.hasNext){
        rowIterator.next();
    }
}
```

Reply:
Just for three lines? That's a bit of an overkill :)
Done